pkg/kube/inject/webhook.go

```diff
@@ -835,9 +835,40 @@ func (wh *Webhook) serveInject(w http.ResponseWriter, r *http.Request) {
 // parseInjectEnvs parse new envs from inject url path
 // follow format: /inject/k1/v1/k2/v2, any kv order works
 // eg. "/inject/cluster/cluster1", "/inject/net/network1/cluster/cluster1"
+// since character slash is used as separator to specify keys and values,
+// if a value needs to contain slashes such as an url, then the path can be
+// created using string `:ENV:` as the separator to include multiple
+// environment variables, the name value pair should be separated by equal sign.
+// Multiple variables should be separated by `:ENV:` eg.
+// "/inject/:ENV:net=network1:ENV:cluster=cluster1:ENV:rootpage=/foo/bar"
+// The above example defines 3 variables, net, cluster and rootpage. Their
+// values are network1, cluster1 and /foo/bar.
 func parseInjectEnvs(path string) map[string]string {
 	path = strings.TrimSuffix(path, "/")
-	res := strings.Split(path, "/")
+	res := func(path string) []string {
+		parts := strings.SplitN(path, "/", 3)
+		// The 3rd part has to start with separator :ENV:
+		// If not, this inject path is considered using slash as separator
+		// If length is less than 3, then the path is simply "/inject",
+		// process just like before :ENV: separator is introduced.
+		if len(parts) == 3 && strings.HasPrefix(parts[2], ":ENV:") {
+			var newRes []string
+			newRes = append(newRes, parts[0], parts[1])
```

Comment:
Putting parts 0 and 1 back seems pointless. Why not just change the following code to not expect them?
```suggestion
```

Reply:
try to avoid changing the logic below. Without adding the two parts, the logic below will need to be changed. Unless we want to complete remove that prefix at the beginning, then also change the logic when pair name and value together. It will be a bit more change. I do not mind to do that if you insist.

Reply:
Unless I'm missing something, I believe I marked the only two places that need to change, below, and they are trivial.

Reply:
@frankbu The block below (loop) starts with index 2, that part will also need to be changed if we get rid of the first two elements in the list. I was trying not to make many changes, but I think not to include the first two elements are good. I can make changes to that section.